### PR TITLE
Annotations on enums

### PR DESCRIPTION
**Pull request by susliko**
Fixes https://github.com/tree-sitter/tree-sitter-scala/issues/201

Problem
-------
Annotations on Scala 3 enums and their cases are not supported, e.g.
```scala
@A enum E:
  @A("") @B case A

```

Solution
-------
Allow annotations in enum definitions

### CODE REVIEW

File: grammar.js [hunk 1]
```diff
@@ -146,7 +147,8 @@ module.exports = grammar({
       )
     ),
 
-    enum_case_definitions: $ => seq(
+    enum_case_definition: $ => seq(
```
Comment (eed3si9n) [inline grammar.js:150]: Why was this renamed here? Since it's a `seq(...)` wouldn't it make sense to keep it plural?

Comment (susliko) [reply grammar.js:150]: Hm
I read it as a "definition of a single enum case", hence i'd prefer it singular 🤔 
One might think that this rule handles definitions of all enum cases
EBNF calls it just EnumCase

Comment (eed3si9n) [reply grammar.js:150]: These names leak out to be tree-sitter queries in various editors, so unless there's a good reason we should avoid renaming them, and even if we do try to make an alias to keep backward compatibility.

Comment (susliko) [reply grammar.js]: Reverted the rename

**Review by eed3si9n — APPROVED** [review]
Thanks 🙏!
